src/js/utils/parser.ts

```diff
@@ -1,14 +1,14 @@
 import { exists } from 'utils/validator';
-import { isNaN } from 'utils/underscore';
+import { isNaN, isValidNumber } from 'utils/underscore';
 
 // Returns the absolute file path based on a relative filepath, and optional base path
-export function getAbsolutePath(path, base) {
-    if (!exists(base)) {
+export function getAbsolutePath(path: string, base?: string): string {
+    if (!base || !exists(base)) {
```

Comment:
`!base` here might be a redundant check. If `exists(base)===true` I don't think `!!base` can possibly be false in application (theoretically if base is 0 it could be, but I can't think of a case where a base path would be `0`).

Reply:
It is redundant. However, because `base` is an optional parameter and `getAbsoultePath` doesn't know that `exists` checks if the input is undefined, when `base` is used later in this method, it raises an error since it "could be undefined". Not sure if there is a workaround for that.

Reply:
ahhh so if i'm reading this right would we potentially be able to get rid of this extra check when `exists` is typed or no? either way let's discuss in standup and come to an agreement on something - so close to getting this merged 👍